### PR TITLE
Incorrect error message when card ZIP validation fails

### LINKED ISSUE #2602: Incorrect ZIP code error message confuses customers
When an order fails ZIP code validation, there is an error message telling them to update the ZIP code.  Sometimes users update the shipping ZIP code since the error doesn't specify.  It would be great to clarify that the _billing_  zipcode failed validation so it's not confusing to users.

Markup 2021-07-27 at 20 19 56

4178877-zen

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #2602

#### Changes proposed in this Pull Request

When an order fails card ZIP code validation, there is an error message telling them to update the ZIP code. Sometimes users update the shipping ZIP code since the error doesn't specify.

127527283-f185ef21-7342-486c-9a41-94523c26c9fa

With the proposed change the error is changed to `We couldn’t verify the postal code in your billing address. Make sure the information is current with your card issuing bank and try again.` for the notice and `We couldn’t verify the postal code in the billing address. If the issue persists, suggest the customer to reach out to the card issuing bank.` for order notes.

#### Testing instructions

- Open your shop page and add products to the cart.
- Go to the Checkout page and try to pay the order with `4000000000000010` test card.
- Validate the correct error message is displayed on the error notice.
- Validate the correct message is displayed in the Order Notes.

Note:
- If the order is successful [blocking payments that fail postal code validation](https://stripe.com/docs/radar/rules#traditional-bank-checks) needs to be activated for your account.
- Once the above rule is enabled, try again with the `4000000000000010` test card.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/class-wc-payment-gateway-wcpay.php [hunk 1]
```diff
@@ -853,21 +853,41 @@ public function process_payment( $order_id ) {
 			}
 
 			if ( ! empty( $payment_information ) ) {
-				$note = sprintf(
-					WC_Payments_Utils::esc_interpolated_html(
-						/* translators: %1: the failed payment amount, %2: error message  */
+				if ( $e instanceof API_Exception && 'card_error' === $e->get_error_type() && 'incorrect_zip' === $e->get_error_code() ) {
```
Comment (kalessil) [inline includes/class-wc-payment-gateway-wcpay.php:856]: I'd suggest reducing to simplify, so we have original `sprintf` only, but generate the detailed message in if-else block. That would better express the change we are making here and make it easier to extend in the future.

**Review by kalessil — CHANGES_REQUESTED** [review]
Looks great overall, I left a suggestion on how to simplify the changes a little bit.

File: includes/class-wc-payment-gateway-wcpay.php (hunk 1 above)
Comment (mordeth) [reply includes/class-wc-payment-gateway-wcpay.php:861]: Makes complete sense. I have simplified the logic, thanks.

**Review by kalessil — APPROVED** [review]
LGTM and great job here to consolidate the different inputs!

**Pull request merged**